src/azul/terraform.py

```diff
@@ -793,18 +793,50 @@ def tf_config(self, app_name):
         del deployment['lifecycle']
         deployment['triggers'] = {'redeployment': deployment.pop('stage_description')}
 
+        # FIXME: Replace with single_quote() from strings.py
+        #        https://github.com/DataBiosphere/azul/pull/6483
+        def sq(s: str) -> str:
+            assert "'" not in s, s
+            return f"'{s}'"
+
         # Using Terraform to specify the REST API minimum compression size
         # proved to be problematic as it would first make an UpdateRestApi call
         # to set the property, followed by a PutRestApi call with mode=overwrite
         # which would reset the property back to its default value (disabled).
         # Setting this property using AWS API Gateway extensions to the OpenAPI
         # specification works around this issue.
         #
+        # We ran into similar difficulties when using Terraform to configure
+        # default responses for the API, so we use these extensions for that
+        # purpose, too.
+        #
         rest_api = resources['aws_api_gateway_rest_api'][app_name]
         assert 'minimum_compression_size' not in rest_api, rest_api
+        assert 'aws_api_gateway_gateway_response' not in resources, resources
         openapi_spec = json.loads(locals[app_name])
         key = 'x-amazon-apigateway-minimum-compression-size'
         openapi_spec[key] = config.minimum_compression_size
+        openapi_spec['x-amazon-apigateway-gateway-responses'] = {
+            f'DEFAULT_{response_type}': {
+                'responseParameters': {
+                    f'gatewayresponse.header.{k}': v
+                    for k, v in {
+                        # Static value method response header parameters must be
+                        # enclosed within a pair of single quotes.
+                        #
+                        # https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html#mapping-response-parameters
+                        # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-gateway-responses.html
+                        #
+                        'Content-Security-Policy': "'default-src 'self''",  # internal quotes intentionally not escaped
+                        'X-Content-Type-Options': sq('nosniff'),
+                        'X-Frame-Options': sq('DENY'),
+                        'Referrer-Policy': sq('strict-origin-when-cross-origin'),
+                        'Strict-Transport-Security': sq('max-age=63072000; includeSubDomains; preload'),
+                        'X-XSS-Protection': sq('1; mode=block')
+                    }.items()
```

Comment:
Extract the dictionary literal into a variable and apply the quoting function to each value in the comprehension. The quoting function should allow internal quotes, or you don't use a quoting function at all.

Reply:
Since it makes sense in all other cases for the quoting function to disallow single quotes in the value the it is wrapping in single quotes, I opted to not use the quoting function here and instead explain with a brief comment.